content/blog/2021/10/new-architecture-to-ease-kubewarden-administrators-lives.md

```diff
@@ -0,0 +1,257 @@
+      - CHOWN
+    required_drop_capabilities:
+      - NET_ADMIN
+```
+
+For a more in depth dive, have a look at our [architecture
+docs](https://docs.kubewarden.io/architecture.html), and the [Kubewarden CRDs documentation](https://github.com/kubewarden/kubewarden-controller/blob/main/docs/crds/README.asciidoc).
+
+## Ok, but what does this mean for administrators?
+
+Moving from 1 PolicyServer to several increases resiliency and segregation. Now,
+we can take admission reviews from policies and separate them into several
+PolicyServers. While the old architecture was already HA, a noisy
+tenant/namespace or a frequently used policy could bring to a crawl the only
+policy server and break havoc in the Cluster. With the new possibility of
+several PolicyServers, a Kubernetes Operator can isolate policy evaluations per
+tenant/namespace, and run mission critical policies separately, making the whole
+infrastructure more resilient.
+
+The new architecture also validates and mutates PolicyServers and
+ClusterAdmissionPolicies with dedicated admission controllers, for a better UX.
+Which means that administrators can rest comfortably when editing them, as
+catastrophic outcomes (such as all policies being dropped by a misconfigured
+PolicyServer, leading to DOS against the cluster) can never happen. Also,
+ClusterAdmissionPolicies will, if no `spec.policyServer` is defined, bind to the
+PolicyServer named `default`. In addition,
+[Finalizers](https://kubernetes.io/docs/concepts/overview/working-with-objects/finalizers/)
+are now added to all Kubewarden CRs, which ensure orderly deletion by the
+Kubewarden Controller.
+
+The inclusion of validating and mutating webhooks for Kuberwarden CRDs means that
+the controller webhook server needs to be securely hooked up to the Kubernetes
+API. In this case, it means using TLS certificates. We have chosen to integrate
+Kuberwarden with [cert-manager](https://cert-manager.io/), to simplify the
+installation, and our Helm Chart today comes with the option for automatically
+creating and setting up Self-Signed certs, or using your own cert-manager
+Issuer.
+
+For ease of deployment, we have separated the CRDs into its own Helm chart:
+`kubewarden-crds`. This prepares the stack for smoother upgrades in the future.
+The Kubewarden Controller and default policy server stay in the
+`kubewarden-controller` Helm chart.
+
+All of the new changes simplify managing clusters. Which makes managing multiple
+clusters via Fleet more consistent and streamlined.
+
+## A hands-on example
+
+Let's install Kubewarden and secure our cluster against privileged pods with a
+simple policy.
+
+### Installing Kubewarden
+
+Make sure you have [`cert-manager`
+installed](https://cert-manager.io/docs/installation/) and then install the
+`kubewarden-crds` and `kubewarden-controller` chart.
+
+For example:
+```console
+$ kubectl apply -f https://github.com/jetstack/cert-manager/releases/download/v1.5.3/cert-manager.yaml
```

Comment:
We probably want to wait for cert-manager to be ready, and for its webhooks to be registered... I did this back in the time, but didn't try with a new deployment:

```
$ kubectl wait --for=condition=Available deployment --timeout=2m -n cert-manager --all
```

This way, we know that the `helm install` will work, because the CRD's of cert-manager will be properly registered and the webhooks it needs will be running.